migration/migration.go

```diff
@@ -275,23 +255,64 @@ func createOrUpdateCommon(typeName string, ctx context.Context, witr models.Work
 		},
 	}
 
+	return createOrUpdateType(typeName, nil, workItemTypeFields, ctx, witr, db)
+}
+
+func createOrUpdateCommon(typeName string, ctx context.Context, witr models.WorkItemTypeRepository, db *gorm.DB) error {
+	workItemTypeFields := map[string]app.FieldDefinition{}
+	extTypeName := models.SystemPlannerItem
+	return createOrUpdateType(typeName, &extTypeName, workItemTypeFields, ctx, witr, db)
+}
+
+func createOrUpdateType(typeName string, extendedTypeName *string, fields map[string]app.FieldDefinition, ctx context.Context, witr models.WorkItemTypeRepository, db *gorm.DB) error {
 	_, err := witr.Load(ctx, typeName)
 	switch err.(type) {
 	case models.NotFoundError:
-		_, err := witr.Create(ctx, nil, typeName, workItemTypeFields)
+		_, err := witr.Create(ctx, extendedTypeName, typeName, fields)
 		if err != nil {
 			return err
 		}
 	case nil:
-		fmt.Println("Work item type exists, will update/overwrite the fields only ")
-		convertedFields, err := models.TEMPConvertFieldTypesToModel(workItemTypeFields)
+		log.Printf("Work item type %v exists, will update/overwrite the fields only and parentPath", typeName)
+		path := "/"
+		if extendedTypeName != nil {
+			log.Printf("Work item type %v extends another type %v, will copy fields from the extended type", typeName, *extendedTypeName)
+			extendedWit, err := witr.Load(ctx, *extendedTypeName)
+			if err != nil {
+				return err
+			}
+			path = extendedWit.ParentPath
+			if path == "/" {
+				path = path + *extendedTypeName
+			} else {
+				path = path + "/" + *extendedTypeName
+			}
+			//load fields from the extended type
+			err = loadFields(ctx, extendedWit, fields)
+			if err != nil {
+				return err
+			}
+		}
+		convertedFields, err := models.TEMPConvertFieldTypesToModel(fields)
 		if err != nil {
 			return err
 		}
 		jsonArray, err := json.Marshal(convertedFields)
 		jsonString := string(jsonArray[:])
-		q := fmt.Sprintf(`UPDATE work_item_types SET fields='%s' where name = '%s'`, jsonString, typeName)
+		q := fmt.Sprintf(`UPDATE work_item_types SET fields='%s',parent_path='%s' where name = '%s'`, jsonString, path, typeName)
```

Comment:
And one more side effect of not exposing parentPath is that this migrations updates parentPath correctly only if we create/update one level deep extensions. For example PlannerItem->Bug.
Relationships like PlannerItem->Issue->Bug or deeper won't be updated correctly in parenPath.